api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -12,69 +12,216 @@
  * One of Treasury's core features is multi-currency support.
  * This allows economy providers and plugins to use different
  * currencies for different transactions, e.g. a daily reward
  * plugin can award players 'Tokens', but a job plugin
  * can award players 'Dollars'. Facilitates great customisability.
+ *
+ * @author lokka30, Geolykt, MrIvanPlays
+ * @since v1.0.0
  */
-@SuppressWarnings("unused")
-public interface Currency {
+public class Currency {
+
+    /**
+     * Creates a new currency with random {@link UUID} {@code currencyId} and a {@code startingBalance} handler of
+     * always returning 0.
+     *
+     * @param currencyChar          a way to identify the currency via a character e.g. '$' for a dollar. can be null.
+     * @param roundedDigits         to which digit shall we round balances. can specify -1 for no rounding.
+     * @param conversionCoefficient coefficient at which this currency is going to be converted to other currencies.
+     * @param balanceFormatter      a function, giving a human-readable format of a balance.
+     * @param names                 acceptable names of the currency. primary name shall be the first specified.
+     * @return new currency instance
+     */
+    @NotNull
+    public static Currency of(
+            @Nullable Character currencyChar,
+            int roundedDigits,
+            double conversionCoefficient,
+            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
+            @NotNull String... names
+    ) {
+        return new Currency(
+                null,
+                currencyChar,
+                roundedDigits,
+                conversionCoefficient,
+                null,
+                balanceFormatter,
+                names
+        );
+    }
+
+    /**
+     * Creates a new currency.
+     *
+     * @param currencyId            the currency id.
+     * @param currencyChar          a way to identify the currency via a character e.g. '$' for a dollar. can be null.
+     * @param roundedDigits         to which digit shall we round balances. can specify -1 for no rounding.
+     * @param conversionCoefficient coefficient at which this currency is going to be converted to other currencies.
+     * @param startingBalance       a function, giving a starting balance for the inputted player uuid, which can be null.
+     * @param balanceFormatter      a function, giving a human-readable format of a balance.
+     * @param names                 acceptable names of the currency. primary name shall be the first specified.
+     * @return new currency instance
+     */
+    @NotNull
+    public static Currency of(
+            @NotNull UUID currencyId,
+            @Nullable Character currencyChar,
+            int roundedDigits,
+            double conversionCoefficient,
+            @Nullable Function<UUID, Double> startingBalance,
+            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
+            @NotNull String... names
```

Comment:
I forget if it works on varargs, but you may consider making these `@NotNull String @NotNull ... names` instead to signal that null elements are also disallowed.